### ISSUE TITLE
NC 25 Error: Undefined array key 1 and Unsupported operand types: int - string

### ISSUE THREAD

**Issue by AndyXheli** [nextcloud/groupfolders#2093]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Delete files within the gorup folders
2. Go into Recycling bin
3. Select all files and restore
4. all files restore but one and errors out
5. 

### Expected behaviour
Tell us what should happen

### Actual behaviour
Tell us what happens instead

### Server configuration

**Operating system:**
Ubuntu 20.04
**Web server:**
Apache2
**Database:**
MariaBD
**PHP version:**
8.0
**Nextcloud version:** (see Nextcloud admin page)
NC 25 Beta 6
**Group folders version:**
13.0.0-beta.1
**Updated from an older Nextcloud/ownCloud or fresh install:**
Nextcloud
**Where did you install Nextcloud from:**
Nextcloud
**Are you using external storage, if yes which one:** local/s3/smb/sftp/...
No
**Are you using encryption:** yes/no
No
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/Saml/...
LDAP

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
{"reqId":"wRZbXA3b33OVEyr88R0n","level":3,"time":"2022-09-14T16:18:52+00:00","remoteAddr":"10.99.0.65","user":"admin ","app":"PHP","method":"GET","url":"/apps/files_trashbin/preview?fileId=482&file=Central%20Square%2FInstall%20How%20To%20Doc%27s%2Fmaster4.pdf%2Fmaster4.pdf.d1587062367&c=1587062367","message":"Undefined array key 1 at /var/www/nextcloud/apps/groupfolders/lib/Trash/TrashBackend.php#338","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36","version":"25.0.0.11","exception":{"Exception":"Error","Message":"Undefined array key 1 at /var/www/nextcloud/apps/groupfolders/lib/Trash/TrashBackend.php#338","Code":0,"Trace":[{"file":"/var/www/nextcloud/apps/groupfolders/lib/Trash/TrashBackend.php","line":338,"function":"onError","class":"OC\\Log\\ErrorHandler","type":"::"},{"file":"/var/www/nextcloud/apps/files_trashbin/lib/Trash/TrashManager.php","line":112,"function":"getTrashNodeById","class":"OCA\\GroupFolders\\Trash\\TrashBackend","type":"->"},{"file":"/var/www/nextcloud/apps/files_trashbin/lib/Controller/PreviewController.php","line":100,"function":"getTrashNodeById","class":"OCA\\Files_Trashbin\\Trash\\TrashManager","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":225,"function":"getPreview","class":"OCA\\Files_Trashbin\\Controller\\PreviewController","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":133,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/App.php","line":172,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/Route/Router.php","line":298,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/var/www/nextcloud/lib/base.php","line":1047,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/var/www/nextcloud/index.php","line":36,"function":"handleRequest","class":"OC","type":"::"}],"File":"/var/www/nextcloud/lib/private/Log/ErrorHandler.php","Line":92,"CustomMessage":"--"},"id":"6321ff05a9bbf"}

{"reqId":"wRZbXA3b33OVEyr88R0n","level":3,"time":"2022-09-14T16:18:52+00:00","remoteAddr":"10.99.0.65","user":"admin","app":"index","method":"GET","url":"/apps/files_trashbin/preview?fileId=482&file=Central%20Square%2FInstall%20How%20To%20Doc%27s%2Fmaster4.pdf%2Fmaster4.pdf.d1587062367&c=1587062367","message":"OCA\\GroupFolders\\Trash\\TrashManager::getTrashItemByFileId(): Return value must be of type array, bool returned in file '/var/www/nextcloud/apps/groupfolders/lib/Trash/TrashManager.php' line 65","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36","version":"25.0.0.11","exception":{"Exception":"Exception","Message":"OCA\\GroupFolders\\Trash\\TrashManager::getTrashItemByFileId(): Return value must be of type array, bool returned in file '/var/www/nextcloud/apps/groupfolders/lib/Trash/TrashManager.php' line 65","Code":0,"Trace":[{"file":"/var/www/nextcloud/lib/private/AppFramework/App.php","line":172,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/Route/Router.php","line":298,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/var/www/nextcloud/lib/base.php","line":1047,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/var/www/nextcloud/index.php","line":36,"function":"handleRequest","class":"OC","type":"::"}],"File":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","Line":165,"Previous":{"Exception":"TypeError","Message":"OCA\\GroupFolders\\Trash\\TrashManager::getTrashItemByFileId(): Return value must be of type array, bool returned","Code":0,"Trace":[{"file":"/var/www/nextcloud/apps/groupfolders/lib/Trash/TrashBackend.php","line":339,"function":"getTrashItemByFileId","class":"OCA\\GroupFolders\\Trash\\TrashManager","type":"->"},{"file":"/var/www/nextcloud/apps/files_trashbin/lib/Trash/TrashManager.php","line":112,"function":"getTrashNodeById","class":"OCA\\GroupFolders\\Trash\\TrashBackend","type":"->"},{"file":"/var/www/nextcloud/apps/files_trashbin/lib/Controller/PreviewController.php","line":100,"function":"getTrashNodeById","class":"OCA\\Files_Trashbin\\Trash\\TrashManager","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":225,"function":"getPreview","class":"OCA\\Files_Trashbin\\Controller\\PreviewController","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":133,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/App.php","line":172,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/Route/Router.php","line":298,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/var/www/nextcloud/lib/base.php","line":1047,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/var/www/nextcloud/index.php","line":36,"function":"handleRequest","class":"OC","type":"::"}],"File":"/var/www/nextcloud/apps/groupfolders/lib/Trash/TrashManager.php","Line":65},"CustomMessage":"--"},"id":"6321ff05a9a64"}

{"reqId":"GnGE4MRuwd7AbY5Tk7ZA","level":3,"time":"2022-09-14T16:19:09+00:00","remoteAddr":"10.99.0.65","user":"admin","app":"core","method":"MOVE","url":"/remote.php/dav/trashbin/axheli/trash/Update%20Pick%20List.pdf.d1587062361","message":"unable to rename, file does not exists : __groupfolders/trash/4/Update Pick List.pdf.d1663172326","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36","version":"25.0.0.11","data":{"app":"core"},"id":"6321ff05a8b97"}
```
</details>

**Comment by CarlSchwan** [comment]
I cannot reproduce :(

In apps/groupfolders/lib/Trash/TrashBackend.php line 338 could you modify the code to add some debug output:

```
\OC::$server->getLogger()->error('relativePath = ' . $relativePath);
[, $folderId, $nameAndTime] = explode('/', $relativePath);
``` 

And then try again, I'm interested in the 'relativePath' before just before the error occurs

**Comment by andyxheli** [comment]
Yes of course once i get to the office I will try and let you know :)

**Comment by andyxheli** [comment]
@CarlSchwan you want it to look like this correct? 
![image](https://user-images.githubusercontent.com/59488153/191768823-a2b3bae2-63b1-4fe2-bbea-ba5685c1aa04.png)

**Comment by CarlSchwan** [comment]
it should be `\OC::$server` and not `$server`

**Comment by andyxheli** [comment]
![image](https://user-images.githubusercontent.com/59488153/191769494-c539f7fe-f4b8-4456-81f2-89569db0bf5c.png)

**Comment by andyxheli** [comment]
@CarlSchwan heres some log files maybe this will help
[nextcloud (3).log](https://github.com/nextcloud/groupfolders/files/9626696/nextcloud.3.log)

**Comment by andyxheli** [comment]
@CarlSchwan here are some more error logs on this 

[Groupfolder error .log](https://github.com/nextcloud/groupfolders/files/9778945/Groupfolder.error.log)

**Comment by joshtrichards** [comment]
Duplicate of #1998 fixed by #2334
